base/logger_audit.go

```diff
@@ -162,3 +166,15 @@ func (al *AuditLogger) shouldLog(id AuditID, ctx context.Context) bool {
 	}
 	return true
 }
+
+// baseFields returns the fields that are common to all audit events.
+func (al *AuditLogger) baseFields() AuditFields {
+	if al == nil {
+		return make(AuditFields)
+	}
+	fields := make(AuditFields, len(al.globalFields))
+	for k, v := range al.globalFields {
+		fields[k] = v
+	}
```

Comment:
This needs to be a "deep-merge", such that it won't overwrite all of `v` if it's an object.
Capella can be injecting just one field in a nested object to overwrite (e.g. `real_userid.username`

I was thinking we can write a function that recursively (for every `map[string]any`) adds or replaces any leaf properties in the object.